app/models/account.rb

```diff
@@ -334,6 +344,16 @@ def mark_verified!
     def to_h
       { name: @name, value: @value, verified_at: @verified_at }
     end
+
+    private
+
+    def string_limit
+      if account.local?
+        255
+      else
+        2047
```

Comment:
I'm worried that's still a bit small. With custom emoji and/or percent-encoded links, this number can rise pretty quickly.
But at least it should cover the issue of having one <255 chars plain link truncated when federating.